src/services/user/user.service.ts

```diff
@@ -0,0 +1,36 @@
+import { User } from '../../entity/user.entity'
+import { v4 as uuidv4 } from 'uuid'
+import { dataSource as dataSource } from '../../dataSource'
+import { UserAlreadyExistsError } from '../../error/user.error'
+import {IUserService} from "./user.service.interface";
+
+class UserService implements IUserService{
+
+    async subscribeEmail(email: string): Promise<User> {
+        const userRepository = dataSource.getRepository(User)
```

Comment:
you are using userRepository in both methods so create instance of repository using constructor to avoid code duplication